lib/spack/llnl/util/tty/log.py

```diff
@@ -31,6 +32,19 @@
 control = re.compile('(\x11\n|\x13\n)')
 
 
+@contextmanager
+def background_safe():
+    signal.signal(signal.SIGTTOU, signal.SIG_IGN)
+    yield
+    signal.signal(signal.SIGTTOU, signal.SIG_DFL)
+
+
+def is_background():
+    if sys.stdout.isatty():
+        return os.getpgrp() != os.tcgetpgrp(sys.stdout.fileno())
+    return False  # not writing to tty, not background
```

Comment:
(question) I'm not sure I follow: I think a backgrounded process could replace `sys.stdout` with a pipe, as we do in the parent process here (and I assume `sys.stdout.isatty()` would return `False` in that case). I think this should include a docstring to mention the circumstances where the function results are meaningful. It might also make sense to prefix this method name with `_` to indicate it is not intended for use outside of this module.

Reply:
Changed the name to `_is_background_tty` for clarity.